src/action/showOnCall.ts

```diff
@@ -0,0 +1,67 @@
+import { Effect, pipe, Schema } from "effect"
+import { TimelineResponseSchema, RotationSchema } from "../utils/decoder"
+import { getCurrentMonthTimeline } from "../api/timeline"
+import { ParseError } from "effect/ParseResult"
+
+export const getBaseRotations = (rotations: ReadonlyArray<RotationSchema>, email: string): ReadonlyArray<RotationSchema> => {
+  return rotations.filter((rotation: RotationSchema) => {
+    if (rotation.periods && rotation.periods.length > 0) {
+      const userMail = rotation.periods[0].recipient.name
+      return userMail === email
```

Comment:
I already return the type of makeProgram:
```ts
const makeProgram = (email: string): Effect.Effect<(readonly RotationSchema[] | [readonly RotationSchema[], number])[], ParseError | Error, never> => {
  const program = pipe(
    getCurrentMonthTimeline(),
    Effect.flatMap(decodeTimelineResponseSchema),
    Effect.andThen(r => {
      const [base, final] = [r.data.baseTimeline.rotations, r.data.finalTimeline.rotations]
      return [getBaseRotations(base, email), getFinalRotations(final, email)]

    })
  )

  return program;
}
```
Also, you can't define `Effect.Effect<
  never,
  never,
  [ReadonlyArray<RotationSchema>, [ReadonlyArray<RotationSchema>, number]]`
the first argument is the value in case of succeed, the second one is the error and the latest is the requirements of Effect. You need to use the new library specs of Effect.